### ISSUE TITLE
Feature request: add pre-scenario tasks

### ISSUE THREAD

**Issue by ZacSweers** [gradle/gradle-profiler#564]
We have a situation where we'd like to run `clean` _between_ different scenarios, but there doesn't appear to be a hook for this currently

**Comment by mrk-han** [comment]
Would this be like `clear-build-cache-before = SCENARIO`  but `clean-before = SCENARIO` ?

Cause I think `cleanup-tasks = ["clean"]` would be called before every build invocation AND therefore every scenario, and in your case you would want to ONLY clean between scenarios and not before every build invocation

**Comment by ljacomet** [comment]
Sorry for the late reply!

***

Can you clarify why `cleanup-tasks` is insufficient for this?
This seems to be invoked before every scenario: https://github.com/gradle/gradle-profiler/blob/a27c6559dc2b42551f91874d45ddbaf0fa8bd428/src/main/java/org/gradle/profiler/gradle/GradleScenarioInvoker.java#L174

**Comment by ZacSweers** [comment]
Seems that should be called preflight tasks if it runs before. "Cleanup" to me reads as an after-only thing

**Comment by mrk-han** [comment]
agree, or preflight-cleanup or something more verbose would be good. always better to be super verbose with these things imho
